grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -87,6 +87,11 @@ class LocationApiController extends BaseDomainApiController {
                 locations += locationService.getLocations(fields, params, isSuperuser, direction, currentLocation, currentUser, params.locationChooser ? true : false)
             }
         }
+
+        if (params.presentation == "toBaseJson") {
+            locations = locations?.collect { Location location -> location.toBaseJson()}
+        }
+
```

Comment:
if you do it like this, won't it anyway use the `toJson()` anyway while rendering when the Bootstrap will be invoked?

Reply:
so I assume it will invoke `toJson()`, so most of the properties which you got rid of in this line, will be `null` then (but the properties will be there)

Reply:
@kchelstowski After doing this collect with `toBaseJson` we have now a list of maps instead of list of Locations, and it won't be processed by Location Marshaller

Reply:
thanks for the explanation